### PR TITLE
fix(datadog_logs sink): relax required input semantic meanings

### PR DESCRIPTION
**Pull request by neuronull**
The `datadog_logs` sink now does not require a semantic meaning input definition for `message` and `timestamp` fields.

While the Datadog logs intake does handle these fields if they are present, they aren't required.

The only impact is that configurations which enable the Log Namespace feature and use a Source input to this sink which does not itself define a semantic meaning for `message` and `timestamp`, no longer need to manually set the semantic meaning for these two fields through a remap transform.

Existing configurations that utilize the Legacy namespace are unaffected.

### CODE REVIEW

**Review by pront — APPROVED** [review]

**Review by jszwedko — COMMENTED** [review]
Is there a test or two that would make sense to have for this?

**Comment by neuronull** [comment]
> Is there a test or two that would make sense to have for this?

👍  I can add testing

**Comment by neuronull** [comment]
> Is there a test or two that would make sense to have for this?

😅 There are a few different layers this could be tested- which are you thinking @jszwedko ? We could

- unit test the function to validate that the event normalization behaves as expected with/without the semantic meanings defined.
- add a test with the DD logs sink with an input source that does and doesnt set the semantic meaning for these with input events that do and don't contain the paths
- add an integration test that emits logs without a message and timestamp field

**Comment by jszwedko** [comment]
> > Is there a test or two that would make sense to have for this?
> 
> 😅 There are a few different layers this could be tested- which are you thinking @jszwedko ? We could
> 
> * unit test the function to validate that the event normalization behaves as expected with/without the semantic meanings defined.
> * add a test with the DD logs sink with an input source that does and doesnt set the semantic meaning for these with input events that do and don't contain the paths
> * add an integration test that emits logs without a message and timestamp field

I'd be content with the first option you have there.

**Comment by neuronull** [comment]
> I'd be content with the first option you have there.

👍  thanks

FWIW I did separately validate locally sending payloads via curl to the logs intake, without message and timestamp fields and they are indexed.

File: changelog.d/dd_logs_semantic_meaning.fix.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+The `datadog_logs` sink now does not require a semantic meaning input definition for `message` and `timestamp` fields.
+
+While the Datadog logs intake does handle these fields if they are present, they aren't required.
+
+The only impact is that configurations which enable the Log Namespace feature and use a Source input to this sink which does not itself define a semantic meaning for `message` and `timestamp`, no longer need to manually set the semantic meaning for these two fields through a remap transform.
+
+Existing configurations that utilize the Legacy namespace are unaffected.
```
[the remark this answers is not in the document]
Comment (neuronull) [reply changelog.d/dd_logs_semantic_meaning.fix.md:7]: Correct. It's a lot of words when I add that and the second stanza is basically just another way of saying the first stanza so I'm open to abandoning the second stanza (pending in my soon-to-be-pushed commits) altogether.

**Review by jszwedko — APPROVED** [review]
🎉

**Pull request merged**